airbyte-integrations/connectors/source-bigquery/source_bigquery/streams.py

```diff
@@ -500,23 +871,53 @@ def _updated_state(self, current_stream_state: MutableMapping[str, Any], latest_
         latest_record_state = latest_record[self.cursor_field]
         stream_state = current_stream_state.get(self.cursor_field)
         if stream_state:
-            self._cursor = max(latest_record_state, stream_state)
+            max_date = max(datetime.strptime(latest_record_state, '%Y-%m-%dT%H:%M:%S.%f%z'), datetime.strptime(stream_state, '%Y-%m-%dT%H:%M:%S.%f%z'))
+            self._cursor = max_date.isoformat(timespec='microseconds')
         else:
             self._cursor = latest_record_state
-        self.state = {self.cursor_field: self._cursor} 
-        self.logger.info(f"current state of {self.name} is {self.state}")
-
+        self.state = {self.cursor_field: self._cursor}
         if datetime.now() >= self._checkpoint_time + timedelta(minutes=15):
             self.checkpoint(self.name, self.state, self.namespace)
             self._checkpoint_time = datetime.now()
         return self.state
 
+    def _extract_borders(self):
```

Comment:
This seems like duplicate code. Can't we regroup this logic? I fear for the maintenance as every time we will change at one place, we will probably need to change at the other place

Reply:
I agree I will add this method in the parent class so that it can be in one place. I will do this change after I finish implementing all the tests in case there are any regressions due to code refactoring.